blocklylib-core/src/main/java/com/google/blockly/android/CategorySelectorUI.java

```diff
@@ -32,13 +34,16 @@
     /**
      * @param categoryCallback the callback for when the user clicks on a category.
      */
-    void setCategoryCallback(CategorySelectorUI.Callback categoryCallback);
-
+    void setCategoryCallback(@Nullable CategorySelectorUI.Callback categoryCallback);
 
     /**
      * Callback for when the user clicks on a category.
      */
     abstract class Callback {
-        public abstract void onCategoryClicked(FlyoutCategory category);
+        /**
+         * Notifies the system that the user has selected a category, or unselected all categories.
+         * @param category The selected category, or null if unselected.
```

Comment:
This description doesn't describe the current usage. This callback is called when a view representing that category was clicked, not after it is selected/deselected. The callback is responsible for deciding what it should do about the click.